src/aggregation/Aggregator.ts

```diff
@@ -243,3 +244,116 @@ export class LongSumAggregator extends AbstractAggregator<Long> {
         output.writeLong(Long.ZERO);
     }
 }
+
+/** @internal */
+export class DistinctValuesAggregator<R> extends AbstractAggregator<Set<R>> {
+
+    classId = AggregatorConstants.DISTINCT;
+
+    readData(input: DataInput): any {
+        this.attributePath = input.readString();
+        const count = input.readInt();
+        for (let i = 0; i < count; i++) {
+            input.readObject();
+        }
+    }
+
+    writeData(output: DataOutput): void {
+        output.writeString(this.attributePath);
+        output.writeInt(0);
+    }
+}
+
+/** @internal */
+export class CanonicalizingHashSet<R> implements IdentifiedDataSerializable, Set<R> {
```

Comment:
Do you implement Set or extend it? I see that extending it works fine on the playground
https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCoASMFmgY7p7ehCTkVOle4jUMdRLYTqCc8LEZzCZmoNJODPHFZZXVtZYYkAAeRJTInDQS8po+rf40gnjbDKv8LqD2jpbYoACqAEoAMsK7sUmxkGSCc+VVQQuaTwVb1UBrDYULY7PagbgUZLJH6QbYmJAECjuMigZEMVDsJzCFLNXxtajBWCMQTCUIAHieAD4IasFDxhMRIKgGcyAN6KEolVarAAU8kgiHyoCeAEpQPzBYqBIIAgxuNwxQpEDKANwCwUAX0URsUCE4ghoeFAAF5BpAJKB6Zx2IhpD9GSLdYo8AFhSKAMxen1qjUAFgCAFYg6r1QGvcl4DiRWaLRD3Igkqk8HKFYKU-B3AFoPBmCKYBKvUb84Xi6Xs3rijgGh4yF5EInnCD2DRvm4zMx2AxmJB+FT1IkGCC9CIvAAiACiqwYiBq+lnSRxs-ADlQAHJhAA5e2zoIqADy3zIu0EI60XOuqEcClc0itTnbOlGWH0KAQiBhwhXL2EiJtEIgGBwOLdCwbBcNwQQlEAA

If this is the case, we can get rid of lots of boilerplate